include/envoy/network/filter.h

```diff
@@ -335,6 +336,14 @@ class FilterChain {
 
 using FilterChainSharedPtr = std::shared_ptr<FilterChain>;
 
+/**
+ * The filter chains could be marked as draining.
+ */
+class DrainableFilterChain : public FilterChain {
+public:
+  virtual void setDrainClose() PURE;
```

Comment:
What does this actually do? Start the drain sequence? `startDrainSequence()` or similar?

Reply:
More like a pure ` DrainManagerImpl::draining_ = true`. Once it is set, all the network filters(AFAIK only HCM) could poll the factory context for the state that the filter chain is going to be destructed.

Reply:
OK then lets call this `startDraining()` or just `drain()`

Reply:
will fix